### PR TITLE
Routing connector fixes

### PR DESCRIPTION
**Pull request by phliar**
- sorting for []Rule was completely bogus
- tests for the routing connector were woefully inadequate
- lots of weirdness between "default" and "*"
- simplified errors
- made most things package-local instead of exported

### CODE REVIEW

File: connectors/routing/routing_config.go [hunk 1]
```diff
@@ -21,57 +21,165 @@
...
+		scopeIsPattern:  scopeIsPattern,
+		scopePattern:    scopePat,
+		prefixIsPattern: prefixIsPattern,
+		prefixPattern:   prefixPat,
+		canonScope:      nScope,
+		canonPfx:        nPrefix,
+	}, nil
+}
+
+// RouteTo (bad name!) says whether or not this Rule can handle the given scope:prefix.
```
Comment (skywhat) [inline connectors/routing/routing_config.go]: why don't we change the name if it's a bad name

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Changelog
 
 ## v3.4.7 (unreleased)
- - Nothing changed yet
+ - Add String() method to all components of the routing connector
```
Comment (skywhat) [inline CHANGELOG.md]: but this diff is not limited to adding String() now, right?

File: names.go [hunk 3]
```diff
@@ -27,66 +27,52 @@ import (
...
 	namePrefixRegex = regexp.MustCompile("^[a-z_][a-z0-9_.]{0,31}$")
+	nameRegex       = regexp.MustCompile("^[a-z_][a-z0-9_]{0,31}$")
 )
 
-// IsValidNamePrefix checks if a name prefix (a namespace prefixed to all DOSA entity type names)
-// conforms the following rules:
-// 1. name prefix starts with [a-z_]
-// 2. the rest of name prefix can contain only [a-z0-9_.]
-// 3. the length of name prefix must be greater than 0 and less than or equal to 32
+// IsValidNamePrefix checks if a name prefix.
```
Comment (skywhat) [inline names.go]: is this comment completed?

File: connectors/routing/routing_config.go (hunk 1 above)
Comment (phliar) [reply connectors/routing/routing_config.go]: A diff is not ready for reviewing until I add reviewers.....

File: names.go (hunk 3 above)
Comment (phliar) [reply names.go]: :)

**Comment by phliar** [comment]
- made almost everything private
- added test to confirm that `footer < foo*`

**Pull request merged**